src/FormBuilder/resources/js/form-builder/src/blocks/fields/amount/inspector/index.tsx

```diff
@@ -139,58 +164,13 @@ const Inspector = ({attributes, setAttributes}) => {
-                                listStyleType: 'none',
-                                padding: 0,
-                                display: 'flex',
-                                flexDirection: 'column',
-                                gap: '16px',
-                            }}
-                        >
-                            {levels.map((amount, index) => {
-                                return (
-                                    <li
-                                        key={'level-option-inspector-' + index}
-                                        style={{
-                                            display: 'flex',
-                                            gap: '16px',
-                                            justifyContent: 'space-between',
-                                            alignItems: 'flex-end',
-                                        }}
-                                        className={'givewp-donation-level-control'}
-                                    >
-                                        <CurrencyControl
-                                            label={__('Donation amount level', 'give')}
-                                            hideLabelFromVision
-                                            value={amount}
-                                            onValueChange={(value) => {
-                                                const newLevels = [...levels];
-
-                                                newLevels[index] = value;
-                                                setAttributes({levels: newLevels});
-                                            }}
-                                        />
-                                        <DeleteButton
-                                            onClick={() => {
-                                                levels.splice(index, 1);
-                                                setAttributes({levels: levels.slice()});
-                                            }}
-                                        />
-                                    </li>
-                                );
-                            })}
-                        </ul>
-                    )}
-                    <AddButton
-                        onClick={() => {
-                            const newLevels = [...levels];
-                            const lastLevel = newLevels[newLevels.length - 1];
-                            const nextLevel = lastLevel ? lastLevel * 2 : 10;
-
-                            newLevels.push(nextLevel.toString());
-                            setAttributes({levels: newLevels});
+                    <Options
+                        currency={true}
+                        multiple={false}
+                        options={levelsOptions}
+                        setOptions={(options) => {
+                            setNewLevels(levelsOptions);
+                            setLevelsOptions(options);
```

Comment:
Same idea from my previous comment, this should really just be updating the levels & the default level in attributes, not the actual options structure.  When passing to the options prop, that's the time you can convert the levels to options.

Reply:
In fact, it should only change the options structure because this component was designed to use a specific state for updating the options on the screen when the user adds a new option or reorder them.